packages/docusaurus-theme-classic/src/theme/hooks/useScrollPosition.ts

```diff
@@ -25,9 +26,13 @@ const useScrollPosition = (
   ) => void,
   deps = [],
 ): void => {
+  const {isScrollMonitorEnabledRef} = useScrollMonitorContext();
   const lastPositionRef = useRef<ScrollPosition | null>(getScrollPosition());
 
   const handleScroll = () => {
+    if (!isScrollMonitorEnabledRef.current) {
```

Comment:
not sure to understand what this does

Reply:
my guess is it prevents the navbar to collapse, but I feel it' not the right place to do that, because this hook is generic. 

What about moving this to `useHideableNavbar` ?

Reply:
The purpose is such that the `window.scroll` usage in `restoreTop` does not cause the navbar or 'back to top' button to show/hide, which it would do otherwise due to the position changing & scroll event triggering. Since the goal is to give the *appearance* that the page didn't actually scroll, I've added this flag to enable/disable 'monitoring scroll events'.

I had it in `useHideableNavbar` originally, but realised that the 'back to top' arrow button in the bottom right is also affected (image below).

![image](https://user-images.githubusercontent.com/22929669/135312352-0b7e5e4b-5d56-4cd9-ac46-ab221140ab77.png)

It seemed safest to instead block *all* scroll events, with the idea that the scrolling is just a workaround, for which we would rather not consider as having scrolled anything